chainio/txmgr/geometric/geometric.go

```diff
@@ -0,0 +1,598 @@
+	}
+	if params.MaxSendTransactionRetry == 0 {
+		params.MaxSendTransactionRetry = defaultParams.MaxSendTransactionRetry
+	}
+	if params.GetTxReceiptTickerDuration == 0 {
+		params.GetTxReceiptTickerDuration = defaultParams.GetTxReceiptTickerDuration
+	}
+	if params.FallbackGasTipCap == 0 {
+		params.FallbackGasTipCap = defaultParams.FallbackGasTipCap
+	}
+	if params.GasMultiplierPercentage == 0 {
+		params.GasMultiplierPercentage = defaultParams.GasMultiplierPercentage
+	}
+	if params.GasPricePercentageMultiplier == nil {
+		params.GasPricePercentageMultiplier = defaultParams.GasPricePercentageMultiplier
+	}
+	if params.GasTipMultiplierPercentage == 0 {
+		params.GasTipMultiplierPercentage = defaultParams.GasTipMultiplierPercentage
+	}
+}
+
+func NewGeometricTxnManager(
+	ethClient EthBackend,
+	wallet wallet.Wallet,
+	logger logging.Logger,
+	metrics Metrics,
+	params GeometricTxnManagerParams,
+) *GeometricTxManager {
+	fillParamsWithDefaultValues(&params)
+	return &GeometricTxManager{
+		ethClient: ethClient,
+		wallet:    wallet,
+		logger:    logger.With("component", "GeometricTxManager"),
+		metrics:   metrics,
+		params:    params,
+	}
+}
+
+// GetNoSendTxOpts This generates a noSend TransactOpts so that we can use
+// this to generate the transaction without actually sending it
+func (m *GeometricTxManager) GetNoSendTxOpts() (*bind.TransactOpts, error) {
+	ctxWithTimeout, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+	from, err := m.wallet.SenderAddress(ctxWithTimeout)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get sender address: %w", err)
+	}
+	return &bind.TransactOpts{
+		From:   from,
+		NoSend: true,
+		Signer: txmgr.NoopSigner,
+	}, nil
+}
+
+func newTxnRequest(tx *types.Transaction) *txnRequest {
+	return &txnRequest{
+		tx:          tx,
+		requestedAt: time.Now(),
+		txAttempts:  make([]*transaction, 0),
+	}
+}
+
+// Send is used to sign and send a transaction to an evm chain.
+// It does gas estimation and gas price bumping to ensure the transaction gets mined,
+// but it does not do nonce management, so the tx argument must have the correct nonce already set.
+//
+// Send is blocking and safe to call concurrently, so sending multiple txs in parallel is safe.
+func (t *GeometricTxManager) Send(ctx context.Context, tx *types.Transaction) (*types.Receipt, error) {
+	return t.processTransaction(ctx, newTxnRequest(tx))
+}
+
+// ProcessTransaction sends the transaction and runs the monitoring loop which will bump the gasPrice until the tx get
+// included.
+func (t *GeometricTxManager) processTransaction(ctx context.Context, req *txnRequest) (*types.Receipt, error) {
+	t.logger.Debug("new transaction",
+		"nonce", req.tx.Nonce(), "gasFeeCap", req.tx.GasFeeCap(), "gasTipCap", req.tx.GasTipCap(),
+	)
+	t.metrics.IncrementProcessingTxCount()
+	defer t.metrics.DecrementProcessingTxCount()
+
+	var txn *types.Transaction
+	var txID wallet.TxID
+	var err error
```

Comment:
you don't this this since line 195 initializes error?

Reply:
[aaa697a](https://github.com/Layr-Labs/eigensdk-go/pull/284/commits/aaa697ac05ebb492f0f1cba584c8eb907bcf71d8)